### PR TITLE
Allows the user of tuples as transaction names

### PR DESCRIPTION
**Pull request by elijahkim**
Closes https://github.com/Nebo15/sage/issues/29

### CODE REVIEW

**Comment by AndrewDryga** [comment]
Hello, @elijahkim! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/Nebo15/sage/pulls/30). Please check our [documentation](https://docs.ebertapp.io) for more information.

File: lib/sage.ex [hunk 1]
```diff
@@ -58,6 +58,9 @@ defmodule Sage do
             when is_tuple(mfa) and tuple_size(mfa) == 3 and
                    (is_atom(elem(mfa, 0)) and is_atom(elem(mfa, 1)) and is_list(elem(mfa, 2)))
 
+  defguardp is_valid_name(name)
```
Comment (AndrewDryga) [inline lib/sage.ex]: Let's call it `is_stage_name` to make it more consistent with other guards.

**Review by AndrewDryga — APPROVED** [review]

**Comment by AndrewDryga** [comment]
Great job @elijahkim, I have only one comment regarding guard name and we are good to merge :)

**Comment by michalmuskala** [comment]
Is there a reason to limit the names? In `Ecto.Multi` we started limiting the names to just atoms, but quickly relaxed it to allow any term.

**Comment by AndrewDryga** [comment]
@michalmuskala I guess we would do the same, I already can imagine people that would use pretty much anything in a tuple.

@elijahkim can you please remove the checks completely and update the type `@type stage_name :: atom()` to `@type stage_name :: term()`?

**Comment by elijahkim** [comment]
Yep! Pushed the new commit. I left the tuple tests in there to validate that it'll still work but obviously it's not exhaustive. I can add or remove tests as needed for sure! @AndrewDryga

**Comment by elijahkim** [comment]
I'm not sure why the formatter is yelling on ci :(

**Comment by AndrewDryga** [comment]
@elijahkim no worries about the formatting, I'll fix it after merging. Thank you for this small but awesome improvement 👍.

**Pull request merged**